pkg/monitoring/mimir/service.go

```diff
@@ -0,0 +1,161 @@
+package mimir
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/observability-operator/pkg/common/password"
+	"github.com/giantswarm/observability-operator/pkg/common/secret"
+	"github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent"
+)
+
+const (
+	authSecretName               = "mimir-gateway-ingress"
+	authSecretNamespace          = "mimir"
+	mimirSpecificSecretName      = "mimir-basic-auth"
```

Comment:
I'm not sure about the names here.

WDYT about having:
mimirNamespace = "mimir"
mimirApiKey = "mimir-api-key"
mimirIngressAuth = "mimir-ingress-auth"

Reply:
I agree with `mimirNamespace = "mimir"`. 
I'd still like to keep `authSecretName = "mimir-gateway-ingress"` though.
As for `mimirApiKey = "mimir-api-key"` why not :shrug:

Reply:
Can you name it smth Line mimir-gateway-ingress-auth then? And rename authSecretName ? I'm just thinking that auth kind means everything and nothing AT thé same time

Reply:
Would `ingressAuthSecretName = "mimir-gateway-ingress-auth" ` work for you ?

Reply:
Looks good :)